### PR TITLE
Create README.md

### PR DESCRIPTION
**Pull request by jeff-zucker**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Discovery based on Solid Social Agent WebId
+
+## Overview
+
+When an app allows a user to log in to a Solid Identity Provider, the app recieves back an authenticated identifier for the user - their WebId.  The general purpose of this document is to descibe what an app can discover based on the WebId.  More specifc purposes are descibred below.
```
Comment (VirginiaBalseiro) [inline README.md:5]: typos: 
recieves > receives
descibe > describe
specifc > specific
descibred > described

File: README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+A Social Agent - a person, group, or organization - may own one or more WebIds.  Each WebId is a URI that can be dereferenced to point to exactly one document, hereafter called a `profile document`.  The term 'profile document', as used here, means a `Social Agent WebId Document` and does not include WebId documents associated with Software Agents.  
+
+Profile documents may optionally point, through various means, to other documents whose intent is to identify the WebId owner and/or descibe details about their data.  These other documents, whatever predicates are used to indicate them, and however they are named, will be refered to collectively as `extended profile documents`.
```
Comment (VirginiaBalseiro) [inline README.md:9]: typos:

descibe > describe
refered > referred

File: README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Profile documents may optionally point, through various means, to other documents whose intent is to identify the WebId owner and/or descibe details about their data.  These other documents, whatever predicates are used to indicate them, and however they are named, will be refered to collectively as `extended profile documents`.  
+
+The term `profile` will be used here to refer to the data that can be gathered in the profile document and all extended profile documents it points to.
+
+Profiles contain two major types of information : 1) identifiying information such as a person or group's name, purpose, contacts,etc. and 2) organizing information such as what types of data the user has, where it is stored, and who has access to it. This document will focus on the later - the information that an app can use to learn how to interact with the data associated with the Social Agent's WebId.
```
Comment (VirginiaBalseiro) [inline README.md:13]: typo:

later > latter

**Review by VirginiaBalseiro — APPROVED** [review]
Looks great! Found some typos but they don't get in the way of understanding the text.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Discovery based on Solid Social Agent WebId
+
+## Overview
+
+When an app allows a user to log in to a Solid Identity Provider, the app receives back an authenticated identifier for the user - their WebId.  The general purpose of this document is to describe what an app can discover based on the WebId.  More specific purposes are described below.
```
Comment (acoburn) [inline README.md:5]: "authenticated identifier" is unclear. I would use "verifiable token identifiying a user". For instance, OpenID connect will return a signed ID token. OAuth2 will return a signed bearer token, a DID-Auth based mechanism will return something different, etc. The point is that there is a mechanism for verifying that this data hasn't been tampered with.

Comment (acoburn) [reply README.md:5]: If this document is building on the [draft WebID specification](https://www.w3.org/2005/Incubator/webid/spec/identity/), that should be made explicit. And if so, the canonical spelling of the term is `WebID`.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+Profile documents may optionally point, through various means, to other documents whose intent is to identify the WebId owner and/or describe details about their data.  These other documents, whatever predicates are used to indicate them, and however they are named, will be referred to collectively as `extended profile documents`.  
+
+The term `profile` will be used here to refer to the data that can be gathered in the profile document and all extended profile documents it points to.
+
+Profiles contain two major types of information : 1) identifiying information such as a person or group's name, purpose, contacts,etc. and 2) organizing information such as what types of data the user has, where it is stored, and who has access to it. This document will focus on the latter - the information that an app can use to learn how to interact with the data associated with the Social Agent's WebId.
```
Comment (acoburn) [inline README.md:13]:
```suggestion
Profiles contain two major types of information : 1) identifiying information such as a person or group's name, purpose, contacts, etc. and 2) organizing information such as what types of data the user has, where it is stored, and who has access to it. This document will focus on the latter - the information that an app can use to learn how to interact with the data associated with the Social Agent's WebId.
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+There is great variability in what information is found in the profile and which documents the information is stored in.  The [interoperability panel](https://solid.github.io/data-interoperability-panel/specification/) is in the process of creating new organization of the profile information. So, there can be more than one discovery process.  This specification aims to describe the discovery process as it is currently in use.
```
Comment (acoburn) [inline README.md:15]: "in the process of creating new organization" is unclear

File: README.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+The following predicates will be the main foucs of the documention :
+
+* foaf:primaryTopic
+* solid:oidcIssuer
+* solid:account
+* solid:publicTypeIndex
+* solid:privateTypeIndex
+* pim:storage
+* pim:preferencesFile
+* acl:trustedApp (possibly omit this for this spec?)
```
Comment (acoburn) [inline README.md:33]: This predicate is not part of the ACL vocabulary and is not described in the WAC specification. I would recommend omitting it.

File: README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+## Related Resources
```
Comment (acoburn) [inline README.md:35]: Please also refer to the WebID specification: https://www.w3.org/2005/Incubator/webid/spec/identity/

File: README.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+To sum up, this document aims to cover
+
+* Social Agents, not Software Agents
+* Profile documents and extended profile documents
+* Discovery of data structures, not personal/group identifying information
+* Current practices, not future specifications
+
+The following predicates will be the main foucs of the documention :
+
+* foaf:primaryTopic
```
Comment (acoburn) [inline README.md:26]: Possibly also the inverse: `foaf:isPrimaryTopicOf`

Comment (jeff-zucker) [reply README.md:26]: Is that what Inrupt IDP uses to point to extended profile documents?  Any other examples "in the wild"?

File: README.md (hunk 7 above)
Comment (jeff-zucker) [reply README.md:33]: However acl:trustedApp is in use and is kind of central to the current app ecosystem.  Hopefully we can think of a way to document it while pointing to alternatives.

File: README.md (hunk 4 above)
Comment (jeff-zucker) [reply README.md:5]: Will change WebId -> WebID, thanks.  Added WebID spec to related resources.  

In regard to "the app receives back an authenticated identifier" - I really want to focus on the WebID, not the token, would this work for you?

     the app receives back a verifiable identifying token and a WebId

File: README.md (hunk 5 above)
Comment (jeff-zucker) [reply README.md:13]: Sorry, I can't spot the difference between the two.

File: README.md (hunk 4 above)
Comment (acoburn) [reply README.md:5]: Right, the format of the response is beyond the scope of this document. The relevant point is that the data is trusted and that it establishes the WebID of the user. That trust can be established via cryptographic signatures or other forms.

I might prefer something more along the lines of "the Solid identity provider supplies the app with data that establishes the identity of the user in terms of a WebID URI". You might even add something about how "the format of this data and the mechanism by which this data is retrieved and verified is beyond the scope of this document". That would leave the door open to OpenID, SAML, GNAP, etc.

File: README.md (hunk 5 above)
Comment (acoburn) [reply README.md]: missing space before the `etc`. Your version has `,etc.`

File: README.md (hunk 4 above)
Comment (jeff-zucker) [reply README.md]: Okay, I went with your suggested phrasing, thanks.

File: README.md (hunk 5 above)
Comment (jeff-zucker) [reply README.md]: Ah, thanks!

File: README.md (hunk 6 above)
Comment (jeff-zucker) [reply README.md]: I went with "in the process of developing alternative discovery processes."  Let me know if that works for you.

File: README.md (hunk 9 above)
Comment (acoburn) [reply README.md:26]: I have generally seen `foaf:primaryTopic`, but for vocabularies that define an inverse functional property (as is the case here), the semantics are equivalent.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+* solid:privateTypeIndex
+* pim:storage
+* pim:preferencesFile
+* acl:trustedApp (possibly omit this for this spec?)
+* owl:sameAs
+* rdfs:seeAlso
+
+## Related Resources
+* [Minutes of meetings](https://github.com/solid/webid-profile/tree/main/meetings)
+* Prior work : [WebID Specifcation](https://www.w3.org/2005/Incubator/webid/spec/identity/)
```
Comment (acoburn) [inline README.md:40]:
```suggestion
* Prior work : [WebID Draft Specification](https://www.w3.org/2005/Incubator/webid/spec/identity/)
```

It is important to refer to this as a draft (I also fixed a typo in this line)

**Comment by ThisIsMissEm** [comment]
This looks good to me!

Though, I think having the `foaf:primaryTopic` as being recommended or mandatory would be a good idea, as that's used by the SDK to retrieve a "profile" with personal information, additionally I believe `pim:storage` is also used by the SDK. cc @NSeydoux / @acoburn

That is, a small subset of the predicates should likely be required as they greatly aid the development of applications.

**Review by ThisIsMissEm — APPROVED** [review]

**Comment by jeff-zucker** [comment]
> used by the SDK 

Thanks, the SDK and tutorials should definitely inform this document.  And I'm glad someone more familiar with them has an eye on the topic.

>  [`foaf:primaryTopic`, `pim:storage`  and ] a small subset of the predicates should likely be required as they greatly aid the development of applications.

We've begun to address those in the [notes](https://github.com/solid/webid-profile/tree/jeff-zucker-patch-2/notes).  I agree those two are quite important.

**Review by theRealImy — APPROVED** [review]

**Pull request merged**